Plugins/BenchmarkTool/BenchmarkTool+Export.swift

```diff
@@ -254,23 +271,154 @@ extension BenchmarkTool {
                 }
             }
         case .metricP90AbsoluteThresholds:
+            let jsonEncoder = JSONEncoder()
+            jsonEncoder.outputFormatting = [.prettyPrinted, .sortedKeys]
+
             try baseline.results.forEach { key, results in
-                let jsonEncoder = JSONEncoder()
-                jsonEncoder.outputFormatting = [.prettyPrinted, .sortedKeys]
+                let fileName = cleanupStringForShellSafety("\(key.target).\(key.name).p90.json")
 
-                var outputResults: [String: BenchmarkThresholds.AbsoluteThreshold] = [:]
-                results.forEach { values in
-                    outputResults[values.metric.rawDescription] = Int(
-                        values.statistics.histogram.valueAtPercentile(90.0)
-                    )
+                var outputResults: [BenchmarkMetric: BenchmarkThreshold] = [:]
+
+                let wantsRelative = self.wantsRelativeThresholds
+                let wantsRange = self.wantsRangeThresholds
+                let wantsRelativeOrRange = wantsRelative || wantsRange
+
+                /// If it's the first run or if relative/range are not specified, then
+                /// override the thresholds file with the new results we have.
+                if runNumber == 1 || !wantsRelativeOrRange {
+                    for values in results {
+                        outputResults[values.metric] = .absolute(
+                            Int(values.statistics.histogram.valueAtPercentile(90.0))
+                        )
+                    }
```

Comment:
Old path of just replacing the static threshold files with the new benchmark results.

If we have a runNumber of 1 (We need to run the benchmarks multiple times for `range`/`relative` so it can be more than 1) or if the user has simply not requested any `relative`/`range` tolerances, then we do what we always did.